src/net.h

```diff
@@ -95,28 +95,32 @@ CNode* FindNode(const CService& ip);
 CNode* FindNode(const NodeId id); //TODO: Remove this
 bool OpenNetworkConnection(const CAddress& addrConnect, bool fCountFailure, CSemaphoreGrant *grantOutbound = NULL, const char *strDest = NULL, bool fOneShot = false, bool fFeeler = false);
 
-struct ListenSocket {
-    SOCKET socket;
-    bool whitelisted;
-
-    ListenSocket(SOCKET socket, bool whitelisted) : socket(socket), whitelisted(whitelisted) {}
-};
-
 class CConnman
 {
 public:
     CConnman();
     ~CConnman();
     bool Start(boost::thread_group& threadGroup, std::string& strNodeError);
     void Stop();
+    bool BindListenPort(const CService &bindAddr, std::string& strError, bool fWhitelisted = false);
+
 private:
+    struct ListenSocket {
```

Comment:
Just a tip, and not a suggestion to change in this PR right now, you can just say 'struct ListenSocket;' here, and fully define it in the cpp file as 'struct CConnman::ListenSocket { ... };', since I don't think any of the users of this class need the fully materialized definition.